exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java

```diff
@@ -143,6 +144,11 @@ public static final RelOptRule getDirFilterOnScan(OptimizerRulesContext optimize
   }
 
   protected void doOnMatch(RelOptRuleCall call, Filter filterRel, Project projectRel, TableScan scanRel) {
+    if (wasAllPartitionsPruned) {
```

Comment:
Pls see commit #ce509af for the changes.  Since the 'all partitions pruned' is a special case, I had to keep track of this state and propagate it during the new FileSelection creation after partition pruning. Thus, if the first phase directory pruning eliminated all partitions and we re-added 1 for schema purposes, this directory selection will be  eventually seen by the ParquetGroupScan which will examine a flag 'wasAllPartitionsPruned' and decide to retrieve only 1 file from this directory.